cli/cmd/resource.go

```diff
@@ -31,17 +29,27 @@ func init() {
 }
 
 func runResource(cmd *cobra.Command, args []string) {
-	URL := viper.GetString("zabbix.url")
+	URL := viper.GetString("zabbix.URL")
 	key := viper.GetString("zabbix.api_key")
-	a := zabbix.NewAPI(URL, key)
+	a, err := zabbix.NewAPI(URL, key, "", "")
+	if err != nil {
+		log.Error("cannot find needed params.", err)
```

Comment:
You want to return here as well, no?
Or better extract these error-ing operation into a private function (makeResource(client Z) err) that returns an error and delegate `runResource` to it.

Reply:
I want to return there too. Mistakenly skipped that.